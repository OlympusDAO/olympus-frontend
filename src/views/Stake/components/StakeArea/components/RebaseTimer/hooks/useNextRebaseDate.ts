import { useQuery } from "react-query";
import { NetworkId } from "src/constants";
import { STAKING_ADDRESSES } from "src/constants/addresses";
import { parseBigNumber } from "src/helpers";
import { useStaticStakingContract } from "src/hooks/useContract";

export const nextRebaseDateQueryKey = () => ["useNextRebaseDate"];

export const useNextRebaseDate = () => {
  const contract = useStaticStakingContract(STAKING_ADDRESSES[NetworkId.MAINNET], NetworkId.MAINNET);

  const key = nextRebaseDateQueryKey();
  return useQuery<Date, Error>(key, async () => {
    const secondsToRebase = await contract.secondsToNextEpoch();

    const parsedSeconds = parseBigNumber(secondsToRebase, 0);

    return new Date(Date.now() + parsedSeconds * 1000);
  });
};
