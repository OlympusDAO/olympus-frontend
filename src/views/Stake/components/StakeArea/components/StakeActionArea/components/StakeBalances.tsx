import { t } from "@lingui/macro";
import { Accordion, AccordionDetails, AccordionSummary } from "@material-ui/core";
import { ExpandMore } from "@material-ui/icons";
import { DataRow } from "@olympusdao/component-library";
import { BigNumber, BigNumberish } from "ethers";
import { NetworkId } from "src/constants";
import { convertGohmToOhm, formatNumber, nonNullable, parseBigNumber } from "src/helpers";
import {
  useFuseBalance,
  useGohmBalance,
  useGohmTokemakBalance,
  useOhmBalance,
  useSohmBalance,
  useV1SohmBalance,
  useWsohmBalance,
} from "src/hooks/useBalance";
import { useCurrentIndex } from "src/hooks/useCurrentIndex";
import { useTestableNetworks } from "src/hooks/useTestableNetworks";

const DECIMAL_PLACES_SHOWN = 4;

const hasVisibleBalance = (balance?: BigNumber, units: BigNumberish = 9) =>
  balance && parseBigNumber(balance, units) > 9 / Math.pow(10, DECIMAL_PLACES_SHOWN + 1);

const formatBalance = (balance?: BigNumber, units: BigNumberish = 9) =>
  balance && formatNumber(parseBigNumber(balance, units), DECIMAL_PLACES_SHOWN);

export const StakeBalances = () => {
  const ohmBalances = useOhmBalance();
  const sohmBalances = useSohmBalance();
  const gohmBalances = useGohmBalance();
  const wsohmBalances = useWsohmBalance();
  const v1sohmBalances = useV1SohmBalance();
  const gohmFuseBalances = useFuseBalance();
  const gohmTokemakBalances = useGohmTokemakBalance();

  const networks = useTestableNetworks();
  const { data: currentIndex } = useCurrentIndex();

  const sohmTokens = [sohmBalances[networks.MAINNET].data, v1sohmBalances[networks.MAINNET].data];
  const totalSohmBalance = sohmTokens.filter(nonNullable).reduce((res, bal) => res.add(bal), BigNumber.from(0));

  const gohmTokens = [
    gohmBalances[networks.MAINNET].data,
    gohmBalances[NetworkId.ARBITRUM].data,
    gohmBalances[NetworkId.AVALANCHE].data,
    gohmBalances[NetworkId.POLYGON].data,
    gohmBalances[NetworkId.FANTOM].data,
    wsohmBalances[NetworkId.MAINNET].data,
    wsohmBalances[NetworkId.ARBITRUM].data,
    wsohmBalances[NetworkId.AVALANCHE].data,
    gohmFuseBalances[NetworkId.MAINNET].data,
    gohmTokemakBalances[NetworkId.MAINNET].data,
  ];
  const totalGohmBalance = gohmTokens.filter(nonNullable).reduce((res, bal) => res.add(bal), BigNumber.from(0));

  const totalStakedBalance = currentIndex
    ? formatBalance(totalSohmBalance.mul(10 ** 9).add(convertGohmToOhm(totalGohmBalance, currentIndex)), 18)
    : BigNumber.from(0);

  const allBalancesLoaded = sohmTokens.every(Boolean) && gohmTokens.every(Boolean);

  return (
    <>
      <DataRow
        id="user-balance"
        title={t`Unstaked Balance`}
        isLoading={!ohmBalances[networks.MAINNET].data}
        balance={`${formatBalance(ohmBalances[networks.MAINNET].data)} OHM`}
      />

      <Accordion className="stake-accordion" square defaultExpanded>
        <AccordionSummary expandIcon={<ExpandMore className="stake-expand" />}>
          <DataRow
            id="user-staked-balance"
            isLoading={!allBalancesLoaded}
            title={t`Total Staked Balance`}
            balance={`${totalStakedBalance} sOHM`}
          />
        </AccordionSummary>

        <AccordionDetails>
          <DataRow
            indented
            title={t`sOHM`}
            id="user-staked-balance"
            isLoading={!sohmBalances[networks.MAINNET].data}
            balance={`${formatBalance(sohmBalances[networks.MAINNET].data)} sOHM`}
          />

          <DataRow
            indented
            title={t`gOHM`}
            isLoading={!gohmBalances[networks.MAINNET].data}
            balance={`${formatBalance(gohmBalances[networks.MAINNET].data, 18)} gOHM`}
          />

          {hasVisibleBalance(gohmBalances[NetworkId.ARBITRUM].data, 18) && (
            <DataRow
              indented
              title={t`gOHM (Arbitrum)`}
              isLoading={!gohmBalances[NetworkId.ARBITRUM].data}
              balance={`${formatBalance(gohmBalances[NetworkId.ARBITRUM].data, 18)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmBalances[NetworkId.AVALANCHE].data, 18) && (
            <DataRow
              indented
              title={t`gOHM (Avalanche)`}
              isLoading={!gohmBalances[NetworkId.AVALANCHE].data}
              balance={`${formatBalance(gohmBalances[NetworkId.AVALANCHE].data, 18)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmBalances[NetworkId.POLYGON].data, 18) && (
            <DataRow
              indented
              title={t`gOHM (Polygon)`}
              isLoading={!gohmBalances[NetworkId.POLYGON].data}
              balance={`${formatBalance(gohmBalances[NetworkId.POLYGON].data, 18)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmBalances[NetworkId.FANTOM].data, 18) && (
            <DataRow
              indented
              title={t`gOHM (Fantom)`}
              isLoading={!gohmBalances[NetworkId.FANTOM].data}
              balance={`${formatBalance(gohmBalances[NetworkId.FANTOM].data, 18)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmTokemakBalances[NetworkId.MAINNET].data, 18) && (
            <DataRow
              indented
              title={t`gOHM (Tokemak)`}
              isLoading={!gohmTokemakBalances[NetworkId.MAINNET].data}
              balance={`${formatBalance(gohmTokemakBalances[NetworkId.MAINNET].data, 18)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmFuseBalances[NetworkId.MAINNET].data, 18) && (
            <DataRow
              indented
              title={t`gOHM (Fuse)`}
              isLoading={!gohmFuseBalances[NetworkId.MAINNET].data}
              balance={`${formatBalance(gohmFuseBalances[NetworkId.MAINNET].data, 18)} gOHM`}
            />
          )}

          {hasVisibleBalance(v1sohmBalances[networks.MAINNET].data) && (
            <DataRow
              indented
              title={t`sOHM (v1)`}
              isLoading={!v1sohmBalances[networks.MAINNET].data}
              balance={`${formatBalance(v1sohmBalances[networks.MAINNET].data)} sOHM`}
            />
          )}

          {hasVisibleBalance(wsohmBalances[networks.MAINNET].data, 18) && (
            <DataRow
              indented
              title={t`wsOHM`}
              isLoading={!wsohmBalances[networks.MAINNET].data}
              balance={`${formatBalance(wsohmBalances[networks.MAINNET].data, 18)} wsOHM`}
            />
          )}

          {hasVisibleBalance(wsohmBalances[NetworkId.ARBITRUM].data, 18) && (
            <DataRow
              indented
              title={t`wsOHM (Arbitrum)`}
              isLoading={!wsohmBalances[NetworkId.ARBITRUM].data}
              balance={`${formatBalance(wsohmBalances[NetworkId.ARBITRUM].data, 18)} wsOHM`}
            />
          )}

          {hasVisibleBalance(wsohmBalances[NetworkId.AVALANCHE].data, 18) && (
            <DataRow
              indented
              title={t`wsOHM (Avalanche)`}
              isLoading={!wsohmBalances[NetworkId.AVALANCHE].data}
              balance={`${formatBalance(wsohmBalances[NetworkId.AVALANCHE].data, 18)} wsOHM`}
            />
          )}
        </AccordionDetails>
      </Accordion>
    </>
  );
};
