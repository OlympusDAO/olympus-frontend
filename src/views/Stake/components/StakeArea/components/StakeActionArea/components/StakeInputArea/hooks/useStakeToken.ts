import { parseUnits } from "@ethersproject/units";
import { t } from "@lingui/macro";
import { ContractReceipt } from "ethers";
import { useMutation, useQueryClient } from "react-query";
import { useDispatch } from "react-redux";
import { GOHM_ADDRESSES, OHM_ADDRESSES, SOHM_ADDRESSES, STAKING_ADDRESSES } from "src/constants/addresses";
import { useWeb3Context } from "src/hooks";
import { balanceQueryKey, useBalance } from "src/hooks/useBalance";
import { useDynamicStakingContract } from "src/hooks/useContract";
import { useTestableNetworks } from "src/hooks/useTestableNetworks";
import { error as createErrorToast, info as createInfoToast } from "src/slices/MessagesSlice";

export const useStakeToken = (toToken: "sOHM" | "gOHM") => {
  const dispatch = useDispatch();
  const client = useQueryClient();
  const networks = useTestableNetworks();
  const { address, networkId } = useWeb3Context();
  const balances = useBalance(OHM_ADDRESSES);
  const contract = useDynamicStakingContract(STAKING_ADDRESSES, true);

  return useMutation<ContractReceipt, Error, string>(
    async amount => {
      if (!amount || isNaN(Number(amount))) throw new Error(t`Please enter a number`);

      const parsedAmount = parseUnits(amount, 9);

      if (!parsedAmount.gt(0)) throw new Error(t`Please enter a number greater than 0`);

      const balance = balances[networks.MAINNET].data;

      if (!balance) throw new Error(t`Please refresh your page and try again`);

      if (parsedAmount.gt(balance)) throw new Error(t`You cannot stake more than your OHM balance`);

      if (!contract) throw new Error(t`Please switch to the Ethereum network to stake your OHM`);

      if (!address) throw new Error(t`Please refresh your page and try again`);

      const shouldRebase = toToken === "sOHM";

      const transaction = await contract.stake(address, parsedAmount, shouldRebase, true);
      return transaction.wait();
    },
    {
      onError: error => {
        dispatch(createErrorToast(error.message));
      },
      onSuccess: async () => {
        const keysToRefetch = [
          balanceQueryKey(address, OHM_ADDRESSES, networkId),
          balanceQueryKey(address, toToken === "sOHM" ? SOHM_ADDRESSES : GOHM_ADDRESSES, networkId),
        ];

        const promises = keysToRefetch.map(key => client.refetchQueries(key, { active: true }));

        await Promise.all(promises);

        dispatch(createInfoToast(t`Successfully staked OHM`));
      },
    },
  );
};
