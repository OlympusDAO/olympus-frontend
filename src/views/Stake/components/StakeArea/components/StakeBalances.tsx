import { t } from "@lingui/macro";
import { Accordion, AccordionDetails, AccordionSummary } from "@material-ui/core";
import { ExpandMore } from "@material-ui/icons";
import { DataRow } from "@olympusdao/component-library";
import { NetworkId } from "src/constants";
import { DecimalBigNumber } from "src/helpers/DecimalBigNumber/DecimalBigNumber";
import { nonNullable } from "src/helpers/types/nonNullable";
import {
  useFuseBalance,
  useGohmBalance,
  useGohmTokemakBalance,
  useOhmBalance,
  useSohmBalance,
  useV1SohmBalance,
  useWsohmBalance,
} from "src/hooks/useBalance";
import { useCurrentIndex } from "src/hooks/useCurrentIndex";
import { useTestableNetworks } from "src/hooks/useTestableNetworks";

const DECIMAL_PLACES_SHOWN = 4;

const hasVisibleBalance = (balance?: DecimalBigNumber) =>
  balance && balance.toApproxNumber() > 9 / Math.pow(10, DECIMAL_PLACES_SHOWN + 1);

const formatBalance = (balance?: DecimalBigNumber) =>
  balance?.toString({ decimals: DECIMAL_PLACES_SHOWN, trim: false, format: true });

export const StakeBalances = () => {
  const networks = useTestableNetworks();
  const { data: currentIndex } = useCurrentIndex();

  const gohmBalances = useGohmBalance();
  const wsohmBalances = useWsohmBalance();

  const ohmBalance = useOhmBalance()[networks.MAINNET].data;
  const sohmBalance = useSohmBalance()[networks.MAINNET].data;
  const v1sohmBalance = useV1SohmBalance()[networks.MAINNET].data;
  const gohmFuseBalance = useFuseBalance()[NetworkId.MAINNET].data;
  const gohmTokemakBalance = useGohmTokemakBalance()[NetworkId.MAINNET].data;

  const sohmTokens = [sohmBalance, v1sohmBalance];

  const gohmTokens = [
    gohmFuseBalance,
    gohmTokemakBalance,
    gohmBalances[networks.MAINNET].data,
    gohmBalances[networks.ARBITRUM].data,
    gohmBalances[networks.AVALANCHE].data,
    gohmBalances[NetworkId.POLYGON].data,
    gohmBalances[NetworkId.FANTOM].data,
    wsohmBalances[networks.MAINNET].data,
    wsohmBalances[networks.ARBITRUM].data,
    wsohmBalances[networks.AVALANCHE].data,
    gohmBalances[NetworkId.OPTIMISM].data,
  ];

  const totalSohmBalance = sohmTokens
    .filter(nonNullable)
    .reduce((res, bal) => res.add(bal), new DecimalBigNumber("0", 9));

  const totalGohmBalance = gohmTokens
    .filter(nonNullable)
    .reduce((res, bal) => res.add(bal), new DecimalBigNumber("0", 18));

  const totalStakedBalance = currentIndex && formatBalance(totalGohmBalance.mul(currentIndex).add(totalSohmBalance));

  const allBalancesLoaded = sohmTokens.every(Boolean) && gohmTokens.every(Boolean);

  return (
    <>
      <DataRow
        id="user-balance"
        title={t`Unstaked Balance`}
        isLoading={!ohmBalance}
        balance={`${formatBalance(ohmBalance)} OHM`}
      />

      <Accordion className="stake-accordion" square defaultExpanded>
        <AccordionSummary expandIcon={<ExpandMore className="stake-expand" />}>
          <DataRow
            id="user-staked-balance"
            isLoading={!allBalancesLoaded}
            title={t`Total Staked Balance`}
            balance={`${totalStakedBalance} sOHM`}
          />
        </AccordionSummary>

        <AccordionDetails>
          <DataRow
            indented
            title={t`sOHM`}
            id="sohm-balance"
            isLoading={!sohmBalance}
            balance={`${formatBalance(sohmBalance)} sOHM`}
          />

          <DataRow
            indented
            title={t`gOHM`}
            isLoading={!gohmBalances[networks.MAINNET].data}
            balance={`${formatBalance(gohmBalances[networks.MAINNET].data)} gOHM`}
          />

          {hasVisibleBalance(gohmBalances[NetworkId.ARBITRUM].data) && (
            <DataRow
              indented
              title={t`gOHM (Arbitrum)`}
              isLoading={!gohmBalances[NetworkId.ARBITRUM].data}
              balance={`${formatBalance(gohmBalances[NetworkId.ARBITRUM].data)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmBalances[NetworkId.AVALANCHE].data) && (
            <DataRow
              indented
              title={t`gOHM (Avalanche)`}
              isLoading={!gohmBalances[NetworkId.AVALANCHE].data}
              balance={`${formatBalance(gohmBalances[NetworkId.AVALANCHE].data)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmBalances[NetworkId.POLYGON].data) && (
            <DataRow
              indented
              title={t`gOHM (Polygon)`}
              isLoading={!gohmBalances[NetworkId.POLYGON].data}
              balance={`${formatBalance(gohmBalances[NetworkId.POLYGON].data)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmBalances[NetworkId.FANTOM].data) && (
            <DataRow
              indented
              title={t`gOHM (Fantom)`}
              isLoading={!gohmBalances[NetworkId.FANTOM].data}
              balance={`${formatBalance(gohmBalances[NetworkId.FANTOM].data)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmBalances[NetworkId.OPTIMISM].data) && (
            <DataRow
              indented
              title={t`gOHM (Optimism)`}
              isLoading={!gohmBalances[NetworkId.OPTIMISM].data}
              balance={`${formatBalance(gohmBalances[NetworkId.OPTIMISM].data)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmTokemakBalance) && (
            <DataRow
              indented
              title={t`gOHM (Tokemak)`}
              isLoading={!gohmTokemakBalance}
              balance={`${formatBalance(gohmTokemakBalance)} gOHM`}
            />
          )}

          {hasVisibleBalance(gohmFuseBalance) && (
            <DataRow
              indented
              title={t`gOHM (Fuse)`}
              isLoading={!gohmFuseBalance}
              balance={`${formatBalance(gohmFuseBalance)} gOHM`}
            />
          )}

          {hasVisibleBalance(v1sohmBalance) && (
            <DataRow
              indented
              title={t`sOHM (v1)`}
              isLoading={!v1sohmBalance}
              balance={`${formatBalance(v1sohmBalance)} sOHM`}
            />
          )}

          {hasVisibleBalance(wsohmBalances[networks.MAINNET].data) && (
            <DataRow
              indented
              title={t`wsOHM`}
              isLoading={!wsohmBalances[networks.MAINNET].data}
              balance={`${formatBalance(wsohmBalances[networks.MAINNET].data)} wsOHM`}
            />
          )}

          {hasVisibleBalance(wsohmBalances[NetworkId.ARBITRUM].data) && (
            <DataRow
              indented
              title={t`wsOHM (Arbitrum)`}
              isLoading={!wsohmBalances[NetworkId.ARBITRUM].data}
              balance={`${formatBalance(wsohmBalances[NetworkId.ARBITRUM].data)} wsOHM`}
            />
          )}

          {hasVisibleBalance(wsohmBalances[NetworkId.AVALANCHE].data) && (
            <DataRow
              indented
              title={t`wsOHM (Avalanche)`}
              isLoading={!wsohmBalances[NetworkId.AVALANCHE].data}
              balance={`${formatBalance(wsohmBalances[NetworkId.AVALANCHE].data)} wsOHM`}
            />
          )}
        </AccordionDetails>
      </Accordion>
    </>
  );
};
