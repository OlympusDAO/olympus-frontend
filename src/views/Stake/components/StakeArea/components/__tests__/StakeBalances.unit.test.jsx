import { DecimalBigNumber } from "src/helpers/DecimalBigNumber/DecimalBigNumber";
import {
  useFuseBalance,
  useGohmBalance,
  useGohmTokemakBalance,
  useOhmBalance,
  useSohmBalance,
  useV1SohmBalance,
  useWsohmBalance,
} from "src/hooks/useBalance";
import { useCurrentIndex } from "src/hooks/useCurrentIndex";
import { NetworkId } from "src/networkDetails";
import { act, render, screen } from "src/testUtils";

import { StakeBalances } from "../StakeBalances";

jest.mock("src/hooks/useBalance");
jest.mock("src/hooks/useCurrentIndex");

describe("<StakeBalances/>", () => {
  it("should render balances and rebase info", async () => {
    useCurrentIndex.mockReturnValue({ data: new DecimalBigNumber("100", 9) });
    // preload user account with staking tokens across networks
    useFuseBalance.mockImplementation(function () {
      const bals = {};
      bals[NetworkId.MAINNET] = { data: new DecimalBigNumber("11", 18) };
      return bals;
    });
    useSohmBalance.mockImplementation(function () {
      const bals = {};
      bals[NetworkId.MAINNET] = { data: new DecimalBigNumber("12", 18) };
      return bals;
    });
    useGohmBalance.mockImplementation(function () {
      const bals = {};
      bals[NetworkId.MAINNET] = { data: new DecimalBigNumber("131", 18) };
      bals[NetworkId.ARBITRUM] = { data: new DecimalBigNumber("132", 18) };
      bals[NetworkId.AVALANCHE] = { data: new DecimalBigNumber("133", 18) };
      bals[NetworkId.POLYGON] = { data: new DecimalBigNumber("134", 18) };
      bals[NetworkId.FANTOM] = { data: new DecimalBigNumber("135", 18) };
      bals[NetworkId.OPTIMISM] = { data: new DecimalBigNumber("136", 18) };
      return bals;
    });
    useGohmTokemakBalance.mockImplementation(function () {
      const bals = {};
      bals[NetworkId.MAINNET] = { data: new DecimalBigNumber("14", 18) };
      return bals;
    });
    useOhmBalance.mockImplementation(function () {
      const bals = {};
      bals[NetworkId.MAINNET] = { data: new DecimalBigNumber("15", 18) };
      return bals;
    });
    useV1SohmBalance.mockImplementation(function () {
      const bals = {};
      bals[NetworkId.MAINNET] = { data: new DecimalBigNumber("16", 18) };
      return bals;
    });
    useWsohmBalance.mockImplementation(function () {
      const bals = {};
      bals[NetworkId.MAINNET] = { data: new DecimalBigNumber("171", 18) };
      bals[NetworkId.ARBITRUM] = { data: new DecimalBigNumber("172", 18) };
      bals[NetworkId.AVALANCHE] = { data: new DecimalBigNumber("173", 18) };
      return bals;
    });

    const stakeBalances = <StakeBalances />;
    var container;
    await act(async () => {
      ({ container } = render(stakeBalances));
    });
    expect(container).toMatchSnapshot();
    expect(await screen.getByText("Unstaked Balance")).toBeInTheDocument();
    expect(await screen.getByText("Total Staked Balance")).toBeInTheDocument();
    // expect to find the correct token balances on the staking screen
    var bals = {
      "Unstaked Balance": "15.0000 OHM",
      "Total Staked Balance": "134,228.0000 sOHM",
      sOHM: "12.0000 sOHM",
      gOHM: "131.0000 gOHM",
      "gOHM (Arbitrum)": "132.0000 gOHM",
      "gOHM (Avalanche)": "133.0000 gOHM",
      "gOHM (Polygon)": "134.0000 gOHM",
      "gOHM (Fantom)": "135.0000 gOHM",
      "gOHM (Optimism)": "136.0000 gOHM",
      "gOHM (Tokemak)": "14.0000 gOHM",
      "gOHM (Fuse)": "11.0000 gOHM",
      "sOHM (v1)": "16.0000 sOHM",
      wsOHM: "171.0000 wsOHM",
      "wsOHM (Arbitrum)": "172.0000 wsOHM",
      "wsOHM (Avalanche)": "173.0000 wsOHM",
    };
    // screen.debug(undefined, 100000);
    for (const [key, value] of Object.entries(bals)) {
      const section = await screen.getByText(key);
      expect(section).toBeInTheDocument();
      const bal = await screen.getByText(value);
      expect(bal).toBeInTheDocument();
    }
  });
});
