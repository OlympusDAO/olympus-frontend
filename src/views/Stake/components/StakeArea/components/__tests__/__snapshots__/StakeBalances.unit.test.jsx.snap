// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StakeBalances/> should render balances and rebase info 1`] = `
<div>
  <div
    class="DataRow-root data-row css-19fvmib"
  >
    <div
      class="MuiBox-root css-1xhj18k"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-4xvy19-MuiTypography-root"
      >
        Unstaked Balance
      </p>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-4xvy19-MuiTypography-root"
      id="user-balance"
      style="text-align: right;"
    >
      15.0000 OHM
    </p>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-expanded MuiAccordion-gutters stake-accordion css-dppvay-MuiPaper-root-MuiAccordion-root"
  >
    <div
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
      >
        <div
          class="DataRow-root data-row css-19fvmib"
        >
          <div
            class="MuiBox-root css-1xhj18k"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-4xvy19-MuiTypography-root"
            >
              Total Staked Balance
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-4xvy19-MuiTypography-root"
            id="user-staked-balance"
            style="text-align: right;"
          >
            134,228.0000 sOHM
          </p>
        </div>
      </div>
      <div
        class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall stake-expand css-y3v3gx-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 20 20"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="MuiAccordion-region"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
            >
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    sOHM
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  id="sohm-balance"
                  style="text-align: right;"
                >
                  12.0000 sOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    gOHM
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  131.0000 gOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    gOHM (Arbitrum)
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  132.0000 gOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    gOHM (Avalanche)
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  133.0000 gOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    gOHM (Polygon)
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  134.0000 gOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    gOHM (Fantom)
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  135.0000 gOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    gOHM (Optimism)
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  136.0000 gOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    gOHM (Tokemak)
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  14.0000 gOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    gOHM (Fuse)
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  11.0000 gOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    sOHM (v1)
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  16.0000 sOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    wsOHM
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  171.0000 wsOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    wsOHM (Arbitrum)
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  172.0000 wsOHM
                </p>
              </div>
              <div
                class="DataRow-root data-row css-19fvmib"
                style="padding-left: 10px;"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  >
                    wsOHM (Avalanche)
                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1psss7x-MuiTypography-root"
                  style="text-align: right;"
                >
                  173.0000 wsOHM
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
